text/0819-public-router-service.md

```diff
@@ -0,0 +1,242 @@
+
+Ember Engines documentation provided at [ember-engines.com](https://ember-engines.com)
+should be updated with `<LinkTo />` replacing all the references of `<LinkToExternal />` component.
+
+Transition path needs to be published on [ember-engines.com](https://ember-engines.com) containing following steps:
+
+1. Replace use of `<LinkToExternal @route="external-route" />` component
+   with `<LinkTo @route="external-route" />` in every engine codebase.
+2. Users using `ember-engines-router-service` would need to replace any of its usage with
+   the `RouterService` provided by `ember-engines`. This addon was created to solve part of
+   this problem previously, and would likely be updated to become a true polyfill for the ember-engines
+   routing service.
+
+Ember Guides should be updated to rationalize and explain the difference between `app/router.js` and `app/services/router.js`.
+The confusion between those two distinct yet different entities exists even today.
+However, it's less a problem today as there is only one thing named `router.js` can exist in the app today and
+ability to create another such file (even with different purpose and semantics) may be confusing.
+
+## Drawbacks
+
+The removal of assertions within `<LinkTo />` specific to ensuring that external paths are properly mapped means that
+it is now the responsibility of the routing service provided by ember-engines to enforce that encapsulation is not
+broken unintentionally, since a user can now provide any fully qualified path to `<LinkTo />`.
+
+A similar issue arises with the routing service's `transitionTo` method when invoked with a url. For instance:
+
+```ts
+router.transitionTo('/');
+```
+
+Some prior discussion on this point is [viewable here](https://github.com/ember-engines/ember-engines/issues/587#issuecomment-483531420).
+In the case of this RFC, it would similarly become the responsibility of the engine's routing service to prefix such a url with the
+engine's mount path as appropriate, or decide to error, to preserve encapsulation.
+
+In practice, we feel the flexibility and improved DX of this solution outweighs these drawbacks, and intend to
+continue enforcing the same constraints via the engines provided service.
+
+## Alternatives
+
+* Utilize `@ember/legacy-built-in-components` to keep `<LinkToExternal />`
+  a thin wrapper around legacy implementation of `<LinkTo />` component.
+
+  As this may be considered temporary solution to solve Ember 4 compatibility for `ember-engines`,
+  this is not viable long term solution as any changes made to `<LinkTo />` component in the Ember.js codebase
+  will need to be backported to `@ember/legacy-built-in-components`.
+
+* Re-implement `<LinkTo />` functionality in `ember-engine`.
+
+  This is not viable long term solution as any changes made to `<LinkTo />` component in the Ember.js codebase
+  would need to be backported to `ember-engines`.
+
+* Deprecate and remove `externalRoutes` mapping.
```

Comment:
🚨 It breaks the isolation principle around ember-engines, where every engine can access any outside contexts, it going in the opposite direction of code isolation.

Reply:
right, that's why though this is an alternative it's not the suggested path of the RFC

Reply:
Surely, just wanted to highlight and provide a reminder/support to the next reviews 😃